src/merkle/store/tests.rs

```diff
@@ -0,0 +1,492 @@
+    assert_eq!(
+        set.get_node(NodeIndex::new(set.depth(), 1)),
+        store.get_node(set.root(), NodeIndex::new(set.depth() - 1, 1)),
+        "node 1 must be the same for both SparseMerkleTree and MerkleStore"
+    );
+    assert_eq!(
+        set.get_node(NodeIndex::new(set.depth(), 2)),
+        store.get_node(set.root(), NodeIndex::new(set.depth() - 1, 2)),
+        "node 2 must be the same for both SparseMerkleTree and MerkleStore"
+    );
+    assert_eq!(
+        set.get_node(NodeIndex::new(set.depth(), 3)),
+        store.get_node(set.root(), NodeIndex::new(set.depth() - 1, 3)),
+        "node 3 must be the same for both SparseMerkleTree and MerkleStore"
+    );
+
+    // STORE MERKLE PATH MATCHS ==============================================================
+    // assert the merkle path returned by the store is the same as the one in the set
+    let result = store
+        .get_path(set.root(), NodeIndex::new(set.depth() - 1, 0))
+        .unwrap();
+    assert_eq!(
+        LEAVES4[0], result.0,
+        "Value for merkle path at index 0 must match leaf value"
+    );
+    assert_eq!(
+        set.get_path(NodeIndex::new(set.depth(), 0)),
+        Ok(result.1),
+        "merkle path for index 0 must be the same for the MerkleTree and MerkleStore"
+    );
+
+    let result = store
+        .get_path(set.root(), NodeIndex::new(set.depth() - 1, 1))
+        .unwrap();
+    assert_eq!(
+        LEAVES4[1], result.0,
+        "Value for merkle path at index 0 must match leaf value"
+    );
+    assert_eq!(
+        set.get_path(NodeIndex::new(set.depth(), 1)),
+        Ok(result.1),
+        "merkle path for index 1 must be the same for the MerkleTree and MerkleStore"
+    );
+
+    let result = store
+        .get_path(set.root(), NodeIndex::new(set.depth() - 1, 2))
+        .unwrap();
+    assert_eq!(
+        LEAVES4[2], result.0,
+        "Value for merkle path at index 0 must match leaf value"
+    );
+    assert_eq!(
+        set.get_path(NodeIndex::new(set.depth(), 2)),
+        Ok(result.1),
+        "merkle path for index 0 must be the same for the MerkleTree and MerkleStore"
+    );
+
+    let result = store
+        .get_path(set.root(), NodeIndex::new(set.depth() - 1, 3))
+        .unwrap();
+    assert_eq!(
+        LEAVES4[3], result.0,
+        "Value for merkle path at index 0 must match leaf value"
+    );
+    assert_eq!(
+        set.get_path(NodeIndex::new(set.depth(), 3)),
+        Ok(result.1),
+        "merkle path for index 0 must be the same for the MerkleTree and MerkleStore"
+    );
+
+    Ok(())
+}
+
+#[test]
+fn wont_open_to_different_depth_root() {
+    let empty = EmptySubtreeRoots::empty_hashes(64);
+    let a = [Felt::new(1); 4];
+    let b = [Felt::new(2); 4];
+
+    // compute the root for a different depth
+    let mut root = Rpo256::merge(&[a.into(), b.into()]);
+    for depth in (1..=63).rev() {
+        root = Rpo256::merge(&[root, empty[depth]]);
+    }
+    let root = Word::from(root);
```

Comment:
I'm not sure we need to do this elaborate computation here - we probably just want to pick a root which is not in the store, and picking it randomly could work just as well.

Reply:
The only reason to keep this test here is because this specific case was a bug in the previous implementation. Asserting its correctness will prevent regression to that

Reply:
I'm fine keeping it here, but we should explain this in the comments then as someone looking at it without context won't understand why we do this computation rather than picking a random root value.

Reply:
Comment updated!